testsuite/fuzzer/fuzz/fuzz_targets/move/aptosvm_publish_and_run.rs

```diff
@@ -205,21 +234,108 @@ fn filter_modules(input: &RunnableState) -> Result<(), Corpus> {
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 1,
+            ]),
+            ident_str!("code").to_owned(),
+        ),
+        ident_str!("publish_package_txn").to_owned(),
+        vec![],
+        vec![
+            bcs::to_bytes(&metadata_serialized).unwrap(),
+            bcs::to_bytes(&code).unwrap(),
+        ],
+    ))
+}
+
+fn publish_transaction_payload(modules: &[CompiledModule]) -> TransactionPayload {
+    let modules_metadatas: Vec<_> = modules
+        .iter()
+        .map(|cm| ModuleMetadata {
+            name: cm.name().to_string(),
+            source: vec![],
+            source_map: vec![],
+            extension: MoveOption::default(),
+        })
+        .collect();
+
+    let all_immediate_deps: Vec<_> = modules
+        .iter()
+        .flat_map(|cm| cm.immediate_dependencies())
+        .map(|mi| PackageDep {
+            account: mi.address,
+            package_name: mi.name.to_string(),
+        })
+        .collect::<BTreeSet<_>>() // leave only uniques
+        .into_iter()
+        .filter(|c| &c.account != modules[0].address()) // filter out package itself
+        .collect::<Vec<_>>();
+
+    let metadata = PackageMetadata {
+        name: "fuzz_package".to_string(),
+        upgrade_policy: UpgradePolicy::compat(), // TODO: currently does not matter. Maybe fuzz compat checks specifically at some point.
+        upgrade_number: 1,
+        source_digest: "".to_string(),
+        manifest: vec![],
+        modules: modules_metadatas,
+        deps: all_immediate_deps,
+        extension: MoveOption::default(),
+    };
+    let pkg_metadata = bcs::to_bytes(&metadata).expect("PackageMetadata must serialize");
+    let mut pkg_code: Vec<Vec<u8>> = vec![];
+    for module in modules {
+        let mut module_code: Vec<u8> = vec![];
+        module
+            .serialize(&mut module_code)
+            .expect("Module must serialize");
+        pkg_code.push(module_code);
+    }
+    code_publish_package_txn(pkg_metadata, pkg_code)
+}
+
 fn run_case(mut input: RunnableState) -> Result<(), Corpus> {
     tdbg!(&input);
-    AptosVM::set_concurrency_level_once(2);
-    let mut vm = FakeExecutor::from_genesis(&VM, ChainId::mainnet()).set_not_parallel();
 
     // filter modules
     filter_modules(&input)?;
 
     for m in input.dep_modules.iter_mut() {
         // m.metadata = vec![]; // we could optimize metadata to only contain aptos metadata
-        m.version = 6; // others don't matter
-    }
-    for module in input.dep_modules.iter() {
+        m.version = VERSION_MAX;
+
         // reject bad modules fast
-        move_bytecode_verifier::verify_module(module).map_err(|_| Corpus::Keep)?;
+        let mut module_code: Vec<u8> = vec![];
+        m.serialize(&mut module_code).map_err(|_| Corpus::Keep)?;
+        let m_de = CompiledModule::deserialize(&module_code).map_err(|_| Corpus::Keep)?;
```

Comment:
We have a config for deserialization, here it uses default variants. Specifically:
```rust
pub fn deserialize(binary: &[u8]) -> BinaryLoaderResult<Self> {
  let config = DeserializerConfig::new(VERSION_DEFAULT, IDENTIFIER_SIZE_MAX);
  Self::deserialize_with_config(binary, &config)
}
```
So version set above in line 311 is not the same as for deserialization. Would be better to use a config (you can fetch it from StateView if you have access to the storage state), or at least use some default which aligns with the version.

Reply:
great catch, fixed it since I think it will also affect coverage